Document DualView::view_[host|device] (#628)
kokkos/kokkos-core-wiki: docs/source/API/containers/DualView.rst

    .. cppkokkos:member:: t_dev d_view

>        The view instance on the *device*, public access deprecated from Kokkos 4.6 on.

Reviewer: ```suggestion
       The view instance on the *device*, public access deprecated in Kokkos 4.6.
```

    .. cppkokkos:member:: t_host h_view

>        The view instance on the *host*, public access deprecated from Kokkos 4.6 on.

Reviewer: ```suggestion
       The view instance on the *host*, public access deprecated in Kokkos 4.6.
```

           dual_view_type::t_host hostView = DV.view<dual_view_type::t_host::memory_space>();

    .. cppkokkos:function:: const t_host& view_host() const;

>        * Return the host-accessible View. With deprecated code enabled, return the View by value.

Reviewer: ```suggestion
       *  Return the host-accessible View. Returns the View by value with `Kokkos_ENABLE_DEPRECATED_CODE_4=ON`
```

Reviewer 2: I agree I think it is worth mentioning the option `Kokkos_ENABLE_DEPRECATED_CODE_4`.
